packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -283,6 +289,30 @@ Future<vm_service.VmService> setUpVmService(
     });
     registrationRequests.add(vmService.registerService('flutterGetSkSL', 'Flutter Tools'));
   }
+
+  if (getIOSProjectInfo != null) {
+    vmService.registerServiceCallback('flutterGetIOSBuildOptions', (Map<String, Object?> params) async {
+      final XcodeProjectInfo? info = await getIOSProjectInfo();
+      if (info == null) {
+        return <String, Object>{
+          'result': <String, Object>{
+            'type': 'Success',
```

Comment:
Should it be a success if the iOS project details can't be determined?

Reply:
This success is a type of response, that means everything complete without error.
See
https://github.com/dart-lang/sdk/blob/a22e003ee1b9fe5d921536d9401829e9ebdefc2a/pkg/vm_service/lib/src/vm_service.dart#L113